frontend/src/lib/fn/csv.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * Converts a JSON array of objects to CSV format
+ * @param {Array} jsonData - Array of objects to convert
+ * @param {string} filename - Name of the file to download (without extension)
+ */
+export const convertJsonToCsv = (jsonData: Record<string, string | number | boolean>[]) => {
+  // Get headers from the first object
+
+  if (jsonData.length === 0) {
+    return new Blob([""], { type: "text/csv;charset=utf-8;" });
+  }
+
+  const headers = Object.keys(jsonData[0]);
+
+  const csvRows = [
+    headers.join(","),
+    // Add all data rows
```

Comment:
nit: the code seems like it doesn't have a lot of moving bits or is complex so the comments feel verbose